mipurifier.h

```diff
@@ -109,19 +109,23 @@ class MiPurifier : public Component, public UARTDevice, public CustomAPIDevice {
       filterlife_sensor->publish_state(atof(val));
     } else if (strcmp(id, "22") == 0) {
       // power (on, off)
-      power_switch->publish_state(strcmp(val, "true") == 0);
+      if (strcmp(val, "true") == 0) {
+        XiaomiPurifierFan->turn_on();
+      } else {
+        XiaomiPurifierFan->turn_off();
+      }
     } else if (strcmp(id, "25") == 0) {
       // mode (auto, night, manual, preset)
       is_preset = false;
       switch (atoi(val)) {
         case 0:
-          mode_select->publish_state("auto");
+          XiaomiPurifierFan->preset_mode = "auto";
```

Comment:
Note: changing mode using physical button on under the LCD does not work correctly.

When doing something like 
```cpp
auto call = XiaomiPurifierFan->make_call();
call.set_preset_mode("auto");
call.perform();
```
it is updated correctly, but for some reason this does not allow for turning the purifier off, it instantly turns on again.

@jaromeyer maybe you'll figure it out, for now I'm leaving this as a draft for others :)

Reply:
Resolved with 

```cpp
    on_preset_set:
      - lambda: |-
          auto c = static_cast<MiPurifier *>(mipurifier);
          if (!id(XiaomiPurifierFan).state) {
            c->set_mode(x);
            c->turn_off();
          } else {
            c->set_mode(x);
          }
```